CRM/Contribute/BAO/Query.php

```diff
@@ -147,12 +130,6 @@ public static function select(&$query) {
       $query->_tables['contribution_payment_instrument'] = 1;
     }
 
-    if (!empty($query->_returnProperties['check_number'])) {
```

Comment:
This can be handled in the main loop based on metadata due to the removal of skipping that loop for 'contribution_' (above) & adding metadata for the field to resolve the ambiguity between check_number & contribution_check_number